src/drivers/spektrum_rc/spektrum_rc.cpp

```diff
@@ -89,17 +90,35 @@ void task_main(int argc, char *argv[])
 	uint16_t raw_rc_values[input_rc_s::RC_INPUT_MAX_CHANNELS];
 	uint16_t raw_rc_count = 0;
 
+#if !defined(__PX4_QURT)
+	pollfd fds[1];
+	fds[0].fd = uart_fd;
+	fds[0].events = POLLIN;
+#endif
+
 	// Main loop
 	while (!_task_should_exit) {
 
-		int newbytes = ::read(uart_fd, &rx_buf[0], sizeof(rx_buf));
+#if !defined(__PX4_QURT)
+		int ret = poll(fds, sizeof(fds) / sizeof(fds[0]), -1);
```

Comment:
and you wouldn't need poll() if you consider the file descriptor as blocking... so you wouldn't need the different implementations